src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerAgreementLevelsControl/MapVisualizerAgreementLevelsControl.tsx

```diff
@@ -1,85 +1,111 @@
+    const agreementLayerKey = 'Agreement'
 
-      await axios.get<Layer>(uri).then((response) => {
-        const { mapId } = response.data
-        const agreementLayerKey = 'Agreement'
-        mapControllerRef.current.removeEarthEngineLayer(agreementLayerKey)
-        mapControllerRef.current.addEarthEngineLayer(agreementLayerKey, mapId)
-      })
-    },
-    [forestOptions.selected, forestOptions.hansenPercentage]
-  )
+    // If any of the dependencies changes and there is an existing agreement layer on the
+    // map, the layer is no longer valid, so remove it. If there is no existing agreement
+    // layer, it's still safe to call `removeEarthEngineLayer`, it'll just do nothing and
+    // return `false`.
+    mapControllerRef.current.removeEarthEngineLayer(agreementLayerKey)
 
-  return (
-    <div className="geo-map-menu-data-visualizer-agreement-levels-control">
-      <p>
-        <strong>Choose the agreement level between all map layers</strong>
-        <br />
-        <br />
-        <small>
-          Agreement level &rdquo;N&rdquo; means that at least N of selected data sources need to agree that a certain
-          pixel is forest area
-        </small>
-      </p>
-      <div className="geo-map-menu-data-visualizer-agreement-levels-boxes">
-        {boxes.map((box, i) => (
-          <button
-            type="button"
-            key={box.title}
-            className={
-              box.disabled
-                ? 'geo-map-menu-data-visualizer-agreement-levels-box disabled'
-                : 'geo-map-menu-data-visualizer-agreement-levels-box'
-            }
-            aria-disabled={box.disabled}
-            onClick={() => handleAgreementLevelSelection(i + 1)}
-          >
-            {box.disabled ? <Icon name="alert" /> : <Icon name={box.checked ? 'remove' : 'circle-add'} />}
-            <br />
-            <p>{box.title}</p>
-          </button>
-        ))}
+    // If the agreement level is greater than the number of selected layers, reset the
+    // agreement state.
+    if (forestOptions.agreementLevel > forestOptions.selected.length) {
+      dispatch(GeoActions.setAgreementLayerSelected(false))
+      dispatch(GeoActions.setAgreementLevel(1))
+      return
+    }
+
+    // If the agreement layer is not selected, don't render anything.
+    if (!forestOptions.agreementLayerSelected) {
+      return
+    }
+
+    // Otherwise, fetch the new agreement layer and add it to the map.
+
+    const layerQuery = forestOptions.selected.map(({ key }) => `&layer=${key}`).join('')
+    const agreementLevelQuery = `&gteAgreementLevel=${forestOptions.agreementLevel}`
+    const hansenQuery = forestOptions.selected.some(({ key }) => key === ForestSource.Hansen)
+      ? `&gteHansenTreeCoverPerc=${forestOptions.hansenPercentage}`
+      : ''
+    const uri = `/api/geo/layers/forestAgreement/?countryIso=FIN${layerQuery}${agreementLevelQuery}${hansenQuery}`
+
+    // TODO: cache the mapId to reduce server calls?
```

Comment:
any chance we can use(or reuse) redux? we can discuss this in our next meeting, now we can even start merging this. what do you think?

Reply:
@minotogna Yes, that's what I thought. We have a simple Redux-based cache for the other layers (`forestOptions.fetchedLayers`), and a similar one could be used here.

@eraviolo That's an interesting point. Could the ids be even cached server-side? Though I don't know about the reliability because the persistence is not guaranteed 🤔

But yeah, let's talk about this in a future meeting. I don't think it's part of the scope of this PR anyway.